cub/cub/thread/thread_reduce.cuh

```diff
@@ -43,15 +48,62 @@
 #  pragma system_header
 #endif // no system header
 
-#include <cub/detail/type_traits.cuh>
-#include <cub/thread/thread_operators.cuh>
+#include <cub/detail/type_traits.cuh>        // are_same_v
+#include <cub/thread/thread_operators.cuh>  // DpxMin
+#include <cub/util_namespace.cuh>
+#include <cub/util_type.cuh>
 
 CUB_NAMESPACE_BEGIN
 
 /// Internal namespace (to prevent ADL mishaps between static functions when mixing different CUB installations)
 namespace internal
 {
 
+//----------------------------------------------------------------------------------------------------------------------
+
```

Comment:
Remark: IMO those section comments are of no help and only add noise. But others my disagree.